manifests/logical_volume.pp

```diff
@@ -2,73 +2,84 @@
 #
 define lvm::logical_volume (
   $volume_group,
-  $size                             = undef,
-  $initial_size                     = undef,
-  Enum['absent', 'present'] $ensure = present,
-  $options                          = 'defaults',
-  $pass                             = '2',
-  $dump                             = '0',
-  $fs_type                          = 'ext4',
-  $mkfs_options                     = undef,
-  Stdlib::Absolutepath $mountpath   = "/${name}",
-  Boolean $mountpath_require        = false,
-  Boolean $mounted                  = true,
-  Boolean $createfs                 = true,
-  $extents                          = undef,
-  $stripes                          = undef,
-  $stripesize                       = undef,
-  $readahead                        = undef,
-  $range                            = undef,
-  $size_is_minsize                  = undef,
-  $type                             = undef,
-  $thinpool                         = false,
-  $poolmetadatasize                 = undef,
-  $mirror                           = undef,
-  $mirrorlog                        = undef,
-  $no_sync                          = undef,
-  $region_size                      = undef,
-  $alloc                            = undef,
+  $size                                               = undef,
+  $initial_size                                       = undef,
+  Enum['absent', 'present'] $ensure                   = present,
+  $options                                            = 'defaults',
+  $pass                                               = '2',
+  $dump                                               = '0',
+  $fs_type                                            = 'ext4',
+  $mkfs_options                                       = undef,
+  Variant[Enum['/'], Stdlib::Absolutepath] $mountpath = "/${name}",
```

Comment:
Yes, but Puppet 3 users should have modules pinned to version tag or commit ref (actually, everybody should, at least in prod) and Puppet 3 hasn't been supported for quite some time. :D